crates/subspace-farmer/src/farmer_cache.rs

```diff
@@ -530,8 +469,7 @@ where
         debug!(%last_segment_index, "Identified last segment index");
 
         let limit = caches
-            .backends
-            .iter()
+            .backends()
```

Comment:
Here a useful method name would be `.total_capacity()` rather than returning all backend as an iterator, you don't need to literally create methods for all individual operations, that is not the goal of an abstraction.